### PR TITLE
test: snapshot/approval test for generated output (#208)

### LINKED ISSUE #208: [Maintenance] testing: Snapshot / approval testing for stable outputs (Verify framework)
## Scope

Where the library produces structured output (formatted strings, JSON, XML, CSV), snapshot testing via the [Verify](https://github.com/VerifyTests/Verify) framework catches accidental format drift that unit tests targeted assertions miss. The first run records the output as a `.verified.txt` file; subsequent runs diff against it.

## Acceptance criteria

- Verify package referenced in the test project.
- Snapshot tests cover every public method whose output is text-shaped, complex-shaped (records, anonymous objects), or has interesting edge-case formatting (culture-sensitive, timezone-sensitive, etc.).
- Snapshot files committed under `tests/.../Snapshots/`.
- CI verifies snapshots match; `.received.txt` files committed during local iteration are gitignored.

## Notes

- Tier-1 follow-up. Only meaningfully valuable where the library produces inspectable output; pure-data utility libraries can close as not-applicable.
- Originated as part of the IAsyncEnumerable-Extensions thorough-review pass; tracked fleet-wide.
- Board: https://github.com/users/Chris-Wolfgang/projects/6

### PR DESCRIPTION
**Pull request by Chris-Wolfgang**
## Summary
Adds a **golden-snapshot test** of what `dotnet new cwconsole` generates — without standing up a test project (that's #22, deferred), so it fits the repo as-is.

- **`tests/snapshots/snapshot.sh`** — packs + installs the template, generates a project with a fixed name, and builds a **normalized per-file hash manifest** (the copyright year is normalized so the snapshot doesn't churn every January). `--check` (default) diffs against the golden and fails on drift; `--update` regenerates the golden after an intended change.
- **`tests/snapshots/cwconsole.manifest`** — the committed golden (19 files).
- **`.github/workflows/snapshot.yml`** — runs `--check` on changes to the template content/csproj/snapshot, push to `main`, and on demand.

## Why it's worth having
The smoke matrix proves the generated app *builds and runs*. This proves the generated **content and structure** don't drift — catching changes that still compile (a tweaked config value, a removed comment, an added/renamed file) that the build can't see.

Verified: `--check` passes on the golden, **bites on drift** (negative-tested), the script is **shellcheck-clean**, and the workflow passes **actionlint + zizmor**. Base: `vNext`.

Closes #208

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/snapshots/snapshot.sh [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+echo "Packing + installing cwconsole..."
+dotnet pack "$PACK_PROJECT" -c Release -o "$work/pkg" > /dev/null
+dotnet new install "$work"/pkg/*.nupkg --force > /dev/null
+
+echo "Generating project..."
+mkdir -p "$work/gen"
+# Ignore a nonzero exit here (the template's "open Instructions.md" post-action
+# is unsupported headless and returns 127); the manifest comparison is the real
+# check and will catch a genuine generation failure as a diff.
+( cd "$work/gen" && dotnet new cwconsole -n "$APP_NAME" > /dev/null 2>&1 ) || true
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply tests/snapshots/snapshot.sh:52]: Addressed in 2cf9284: excluded gitignored `logs/` from the manifest (that was the CI failure - my local golden had runtime logs a clean checkout doesn't), binary-safe hashing for the icons (raw bytes, not through sed), `cleanup()` no longer uninstalls a pre-existing global cwconsole install, fixed the prerequisite comment, and the push trigger now covers the app csproj. Golden regenerated (19 files); CI is green.

File: tests/snapshots/snapshot.sh [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+# Build the manifest. Normalize the one dynamic value the template injects: the
+# copyright year (a `now` generator), so the snapshot doesn't churn every January.
+manifest="$work/manifest.txt"
+: > "$manifest"
+while IFS= read -r -d '' f; do
+  rel="${f#"$work/gen/"}"
+  hash="$(sed -E 's/(Copyright )[0-9]{4}/\1YYYY/g' "$f" | sha256sum | cut -d' ' -f1)"
+  printf '%s  %s\n' "$hash" "$rel" >> "$manifest"
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply tests/snapshots/snapshot.sh]: Addressed in 2cf9284: excluded gitignored `logs/` from the manifest (that was the CI failure - my local golden had runtime logs a clean checkout doesn't), binary-safe hashing for the icons (raw bytes, not through sed), `cleanup()` no longer uninstalls a pre-existing global cwconsole install, fixed the prerequisite comment, and the push trigger now covers the app csproj. Golden regenerated (19 files); CI is green.

File: tests/snapshots/snapshot.sh [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+# Packs + installs the cwconsole template, generates a project with fixed inputs,
+# and produces a normalized manifest (one `sha256␠relative/path` line per file,
+# sorted). Compares it to the committed golden so any *content or structure* drift
+# in what the template generates fails CI — even changes that still compile, which
+# the build-based smoke matrix can't see.
+#
+#   ./snapshot.sh --check    (default) regenerate and diff against the golden; nonzero on drift
+#   ./snapshot.sh --update             regenerate and overwrite the golden (run after an intended change)
+#
+# Runs on Linux/macOS and Windows git-bash (needs: dotnet, unzip-free, sha256sum).
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply tests/snapshots/snapshot.sh]: Addressed in 2cf9284: excluded gitignored `logs/` from the manifest (that was the CI failure - my local golden had runtime logs a clean checkout doesn't), binary-safe hashing for the icons (raw bytes, not through sed), `cleanup()` no longer uninstalls a pre-existing global cwconsole install, fixed the prerequisite comment, and the push trigger now covers the app csproj. Golden regenerated (19 files); CI is green.

File: tests/snapshots/cwconsole.manifest [hunk 4]
```diff
@@ -0,0 +1,21 @@
+04c60b0105a8099e11e044601f2c7ec27bf2bf92a792554b8d8c48c98e33d700  AppSettings.json
+2d1d8c986340c3a3d89dd9261d8bc9746ee1c20d3ad6999ee1b5d9d169516f6e  SnapshotApp.csproj
+45ce616c732f8aeebbaa93c6ef54c735f1ae139e5f047747f48ad02bc29d7619  ReadMe.md
+4cf7061ff7a551c65529629f4222e9c7a9420f76923c8892c74b1a98898b1d90  logs/log-20260709.txt
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply tests/snapshots/cwconsole.manifest]: Addressed in 2cf9284: excluded gitignored `logs/` from the manifest (that was the CI failure - my local golden had runtime logs a clean checkout doesn't), binary-safe hashing for the icons (raw bytes, not through sed), `cleanup()` no longer uninstalls a pre-existing global cwconsole install, fixed the prerequisite comment, and the push trigger now covers the app csproj. Golden regenerated (19 files); CI is green.

File: tests/snapshots/snapshot.sh [hunk 5]
```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
+#
+# Snapshot (approval) test for the generated project output (issue #208).
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply tests/snapshots/snapshot.sh:3]: Used a script-based golden snapshot rather than a Verify (.verified/.received) test on purpose: this repo has no test project (that's #22, deferred), and standing one up just for snapshots is heavier than the drift-detection value. The script gives the same guarantee - normalized per-file manifest, --check in CI, --update to approve - without the test-harness dependency. If #22 lands, this can migrate to Verify.

File: tests/snapshots/snapshot.sh [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+echo "Packing + installing cwconsole..."
+dotnet pack "$PACK_PROJECT" -c Release -o "$work/pkg" > /dev/null
+dotnet new install "$work"/pkg/*.nupkg --force > /dev/null
+
+echo "Generating project..."
+mkdir -p "$work/gen"
+# Ignore a nonzero exit here (the template's "open Instructions.md" post-action
+# is unsupported headless and returns 127); the manifest comparison is the real
+# check and will catch a genuine generation failure as a diff.
+( cd "$work/gen" && dotnet new cwconsole -n "$APP_NAME" > /dev/null 2>&1 ) || true
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply tests/snapshots/snapshot.sh:52]: Fixed in 54a763a — the script now sets `DOTNET_CLI_HOME` to a throwaway dir under $work, so `dotnet new install` uses an isolated template hive and never overwrites (or needs to uninstall) a contributor's global cwconsole install. Verified the global `dotnet new` hive is untouched before/after a run.

File: tests/snapshots/snapshot.sh [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+#   unsafe and gives platform-dependent results.
+manifest="$work/manifest.txt"
+: > "$manifest"
+while IFS= read -r -d '' f; do
+  rel="${f#"$work/gen/"}"
+  if grep -Iq . "$f"; then
+    hash="$(sed -E 's/(Copyright )[0-9]{4}/\1YYYY/g' "$f" | sha256sum | cut -d' ' -f1)"
+  else
+    hash="$(sha256sum "$f" | cut -d' ' -f1)"
+  fi
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply tests/snapshots/snapshot.sh:69]: Fixed in 54a763a — added a portable `sha256()` helper that uses `sha256sum` when present and falls back to `shasum -a 256` (macOS/BSD), so the 'runs on Linux/macOS' claim holds.

File: tests/snapshots/cwconsole.manifest [hunk 8]
```diff
@@ -0,0 +1,19 @@
+04c60b0105a8099e11e044601f2c7ec27bf2bf92a792554b8d8c48c98e33d700  AppSettings.json
+2d1d8c986340c3a3d89dd9261d8bc9746ee1c20d3ad6999ee1b5d9d169516f6e  SnapshotApp.csproj
+45ce616c732f8aeebbaa93c6ef54c735f1ae139e5f047747f48ad02bc29d7619  ReadMe.md
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply tests/snapshots/cwconsole.manifest:3]: Fixed — updated the PR description to 19 files (the count after excluding the gitignored logs/). The committed golden is the source of truth at 19.
